keps/sig-node/5419-pod-level-resources-in-place-resize/README.md

```diff
@@ -0,0 +1,911 @@
+  // +optional
+  AllocatedResources ResourceList
+}
+```
+
+##### Resize Restart Policy
+
+Pod-level resize policy is not supported in the alpha stage of Pod-level resource
+feature. While a pod-level resize policy might be beneficial for VM-based runtimes
+like Kata Containers (potentially allowing the hypervisor to restart the entire VM
+on resize), this is a topic for future consideration. We plan to engage with the
+Kata community to discuss this further and will re-evaluate the need for a pod-level
+policy in subsequent development stages.
+
+The absence of a pod-level resize policy means that container restarts are
+exclusively managed by their individual `resizePolicy` configs. And, if the
+containers do not specify `resizePolicy`, `PreferNoRestart` is the default resize
+policy. The resizes (pod-level or container-level) will not result in container restarts.
+
+The example below of a pod with pod-level resources demonstrates several key
+aspects of this behavior, showing how containers without explicit limits (which
+inherit pod-level limits) interact with resize policy, and how containers with
+specified resources remain unaffected by
+pod-level resizes.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod-level-resources
+spec:
+  resources:
+    requests:
+      cpu: 100m
+      memory: 100Mi
+    limits:
+      cpu: 200m
+      memory: 200Mi
+  containers:
+    - name: c1
+      image: registry.k8s.io/pause:latest
+      resizePolicy:
+        - resourceName: "cpu"
+          restartPolicy: "NotRequired"
+        - resourceName: "memory"
+          restartPolicy: "RestartRequired"
+    - name: c2
+      image: registry.k8s.io/pause:latest
+      resources:
+        requests:
+          cpu: 50m
+          memory: 50Mi
+        limits:
+          cpu: 100m
+          memory: 100Mi
+      resizePolicy:
+        - resourceName: "cpu"
+          restartPolicy: "NotRequired"
+        - resourceName: "memory"
+          restartPolicy: "RestartRequired"
+```
+
+In this example:
+* CPU resizes: Neither container requires a restart for CPU resizes, and therefore CPU resizes at neither the container nor pod level will trigger any restarts.
+* Container c1 (inherited memory limit): c1 does not define any container level
+  resources, so the effective memory limit of the container is determined by the
+  pod-level limit. When the pod's limit is resized, c1's effective memory limit
+  changes. Because c1's memory resizePolicy is RestartRequired, a resize of the
+  pod-level memory limit will trigger a restart of container c1.
+* Container c2 (specified memory limit): c2 does define container-level resources,
+  so the effective memory limit of c2 is the container level limit. Therefore, a
+  resize of the pod-level memory limit doesn't change the effective container limit,
+  so the c2 is not restarted when the pod-level memory limit is resized.
+
+##### Implementation Details
+
+###### Allocating Pod-level Resources
```

Comment:
Can you add a note that pod-level resize requests will need to be considered when determining the priority of the resize (ref https://github.com/kubernetes/enhancements/pull/5266)?

(I'm planning to include this in my implementation of prioritized resizes so you shouldn't need to do anything else for this)